src/pages/user/wallet.tsx

```diff
@@ -0,0 +1,557 @@
+import { Currency } from '@prisma/client';
+import { useQueryBuzzAccount } from '~/components/CivitaiWrapped/CivitaiSessionProvider';
+import { numberWithCommas } from '~/utils/number-helpers';
+import { CurrencyIcon } from '~/components/Currency/CurrencyIcon';
+import { createServerSideProps } from '~/server/utils/server-side-helpers';
+
+ChartJS.register(CategoryScale, LinearScale, PointElement, LineElement, ChartTooltip);
+
+const options = {
+  aspectRatio: 2.5,
+  plugins: {
+    legend: {
+      display: false,
+    },
+    title: {
+      display: false,
+    },
+  },
+};
+
+export const getServerSideProps = createServerSideProps({
+  useSession: true,
+  resolver: async ({ session }) => {
+    const features = getFeatureFlags({ user: session?.user });
+    if (!features.buzz) {
+      return { notFound: true };
+    }
+  },
+});
+
+const moveBackground = keyframes({
+  '0%': {
+    backgroundPosition: '0% 50%',
+  },
+  '50%': {
+    backgroundPosition: '100% 50%',
+  },
+  '100%': {
+    backgroundPosition: '0% 50%',
+  },
+});
+
+const pulse = keyframes({
+  '0%': {
+    stroke: '#FFD43B',
+    opacity: 1,
+  },
+  '50%': {
+    stroke: '#F59F00',
+    opacity: 0.7,
+  },
+  '100%': {
+    stroke: '#F08C00',
+    opacity: 1,
+  },
+});
+
+const useStyles = createStyles((theme) => ({
+  lifetimeBuzzContainer: {
+    border: `2px solid ${theme.colors.yellow[7]}`,
+    background: theme.fn.linearGradient(45, theme.colors.yellow[4], theme.colors.yellow[1]),
+    animation: `${moveBackground} 5s ease infinite`,
+    backgroundSize: '200% 200%',
+  },
+  lifetimeBuzzBadge: {
+    background: theme.colors.dark[6],
+    borderRadius: '22px',
+    padding: '10px 20px',
+  },
+  tileCard: {
+    backgroundColor: theme.colorScheme === 'dark' ? theme.colors.dark[5] : theme.colors.gray[0],
+  },
+  featureCard: {
+    backgroundColor: theme.colorScheme === 'dark' ? theme.colors.dark[5] : theme.colors.gray[0],
+  },
+  lifetimeBuzz: {
+    animation: `${pulse} 1s ease-in-out infinite`,
+  },
+}));
+
+export default function UserWallet() {
+  const currentUser = useCurrentUser();
+  const theme = useMantineTheme();
+  const { classes, cx } = useStyles();
+
+  const { data: { transactions = [] } = {}, isLoading } = trpc.buzz.getUserTransactions.useQuery({
+    limit: 200,
+  });
+  const { lifetimeBalance = 0 } = useQueryBuzzAccount({
+    enabled: !!currentUser,
+  });
+
+  const transactionsReversed = useMemo(() => [...(transactions ?? [])].reverse(), [transactions]);
+
+  const starterBuzzAmount = (transactions ?? []).reduce((acc, transaction) => {
+    return acc - transaction.amount;
+  }, currentUser?.balance ?? 0);
+
+  const items: Record<string, number> = useMemo(() => {
+    if (!transactions) return {};
+
+    let start = starterBuzzAmount;
+
+    return transactionsReversed.reduce((acc, transaction) => {
+      const updated = {
+        ...acc,
+        [dayjs(transaction.date).format('DD/MM/YYYY')]: start + transaction.amount,
```

Comment:
Maybe we can use the `formatDate` helper?
```suggestion
        [formatDate(transaction.date, 'DD/MM/YYYY')]: start + transaction.amount,
```